url146c819f25_jlink_Parse4SJM_tgz-pJ8-sjm_examples_book_robot_ShowRobotRefactoredJ8/src/sjm/examples/book/tokens/ShowSuppliedReader.java

```diff
@@ -0,0 +1,39 @@
+package sjm.examples.book.tokens;
+
+import java.io.*;
+
+import sjm.parse.tokens.*;
+
+/**
+ * This class shows that you can supply your own reader to
+ * a tokenizer.
+ * 
+ * @author Steven J. Metsker
+ *
+ * @version 1.0 
+ */
+public class ShowSuppliedReader {
+	/**
+	 * Just an example.
+	 */
+	public static void main(String[] args) throws IOException {
+		String s = "Let's file this away.";
+		FileWriter fw = new FileWriter("temp.txt");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fw*. The resource is referenced at line 22. The resource is closed at line 23. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *fw* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).